aws/resource_aws_iam_user.go

```diff
@@ -174,6 +181,31 @@ func resourceAwsIamUserUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("tags") {
+		// Reset all tags to empty set
+		oraw, nraw := d.GetChange("tags")
+		o := oraw.(map[string]interface{})
+		n := nraw.(map[string]interface{})
+		c, r := diffTagsIAM(tagsFromMapIAM(o), tagsFromMapIAM(n))
+
+		_, untagErr := iamconn.UntagUser(&iam.UntagUserInput{
+			UserName: aws.String(d.Id()),
+			TagKeys:  tagKeysIam(r),
+		})
+		if untagErr != nil {
+			return fmt.Errorf("error deleting IAM user tags: %s", untagErr)
+		}
+
+		input := &iam.TagUserInput{
```

Comment:
We should perform length check here to ensure we don't try to call `TagUser` with an empty `Tags`, e.g.

```go
if len(c) > 0 {
  // existing logic
}
```